From: Bhavana Nagendra <bnagendr@redhat.com>
Subject: Re: RHEL5.1 PATCH : SMBus does not work on ATI/AMD SB700 chipset
Date: Thu, 21 Jun 2007 10:06:47 -0400
Bugzilla: 244150
Message-Id: <467A85F7.4030702@redhat.com>
Changelog: [i2c] SMBus does not work on ATI/AMD SB700 chipset


--- linux-2.6.18.x86_64/drivers/i2c/busses/i2c-piix4.c.smorig	2007-06-21 10:03:36.000000000 -0400
+++ linux-2.6.18.x86_64/drivers/i2c/busses/i2c-piix4.c	2007-06-21 10:04:07.000000000 -0400
@@ -23,7 +23,7 @@
    Supports:
 	Intel PIIX4, 440MX
 	Serverworks OSB4, CSB5, CSB6, HT-1000
-	ATI IXP200, IXP300, IXP400, SB600
+	ATI IXP200, IXP300, IXP400, SB600, SB700
 	SMSC Victory66
 
    Note: we assume there can only be one device, with one SMBus interface.
@@ -399,6 +399,8 @@ static struct pci_device_id piix4_ids[] 
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP600_SMBUS),
 	  .driver_data = 0 },
+	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP700_SMBUS),
+	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_OSB4),
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_CSB5),
--- linux-2.6.18.x86_64/include/linux/pci_ids.h.smorig	2007-06-21 10:03:43.000000000 -0400
+++ linux-2.6.18.x86_64/include/linux/pci_ids.h	2007-06-21 10:04:07.000000000 -0400
@@ -368,6 +368,7 @@
 #define PCI_DEVICE_ID_ATI_IXP600_IDE	0x438c
 #define PCI_DEVICE_ID_ATI_IXP700_SATA	0x4390
 #define PCI_DEVICE_ID_ATI_IXP700_IDE	0x439c
+#define PCI_DEVICE_ID_ATI_IXP700_SMBUS	0x4395
 
 #define PCI_VENDOR_ID_VLSI		0x1004
 #define PCI_DEVICE_ID_VLSI_82C592	0x0005
--- linux-2.6.18.x86_64/Documentation/i2c/busses/i2c-piix4.smorig	2007-06-21 10:03:50.000000000 -0400
+++ linux-2.6.18.x86_64/Documentation/i2c/busses/i2c-piix4	2007-06-21 10:04:07.000000000 -0400
@@ -6,7 +6,7 @@ Supported adapters:
     Datasheet: Publicly available at the Intel website
   * ServerWorks OSB4, CSB5, CSB6 and HT-1000 southbridges
     Datasheet: Only available via NDA from ServerWorks
-  * ATI IXP200, IXP300, IXP400 and SB600 southbridges
+  * ATI IXP200, IXP300, IXP400, SB600 and SB700 southbridges
     Datasheet: Not publicly available
   * Standard Microsystems (SMSC) SLC90E66 (Victory66) southbridge
     Datasheet: Publicly available at the SMSC website http://www.smsc.com

